Mathlib/CategoryTheory/Enriched/Limits/HasConicalLimits.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2025 Jon Eugster. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster, Dagur Asgeirsson, Emily Riehl
+-/
+import Mathlib.CategoryTheory.Enriched.Ordinary.Basic
+
+/-!
+# HasConicalLimits
+
+This file contains different statements about the (non-constructive) existence of conical limits.
+
+The main constructions are the following.
+
+- `HasConicalLimit`: there exists a conical limit for `F : J ⥤ C`.
+- `HasConicalLimitsOfShape J`: All functors `F : J ⥤ C` have conical limits.
+- `HasConicalLimitsOfSize.{v₁, u₁}`: For all small `J` all functors `F : J ⥤ C` have conical limits.
+- `HasConicalLimits `: `C` has all (small) conical limits.
+
+## References
+
+* [Kelly G.M., *Basic concepts of enriched category theory*][kelly2005]:
+  See section 3.8 for a similar treatment, although the content of this file is not directly
+  adapted from there.
+-/
+
+universe v₁ u₁ v₂ u₂ w v' v u u'
+
+namespace CategoryTheory.Enriched
+
+open Limits
+
+section Definitions
+
+-- note: for the classes it seems that instance inference wants `V` to be an `outParam`.
+variable {J : Type u₁} [Category.{v₁} J]
+variable (V : outParam <| Type u') [Category.{v'} V] [MonoidalCategory V]
```

Comment:
I would prefer `V` not being an `outParam` (a feature which is essentially not used at all in the category theory API).

Reply:
Could you give an example where the instance inference would not work without the `outParam`?

Reply:
I do not think that `V` is uniquely determined. For example, if `C` is a category of functors to `SSet`, we may use `V := SSet` or `C`.
Let us keep the `outParam` for now.

Reply:
ok, done

Reply:
Actually, I was just suggesting keeping `V` a `outParam`! This way, we may try to see if making `HasConicalLimitsOfShape -> HasLimitsOfShape` an instance leads to issues in future developments or not (and in any case, the fix would be easy...).

Reply:
sorry, misread your comment. I'll update it tomorrow :)

Reply:
done